### PR TITLE
Update compiler test documentation

### PR DESCRIPTION
**Pull request by tromey**
Update the compiler test documentation to document ignore-gdb-version
and min-system-llvm-version; and expand the min-gdb-version,
min-lldb-version, and min-llvm-version documentation a little.

### CODE REVIEW

File: src/test/COMPILER_TESTS.md [hunk 1]
```diff
@@ -53,6 +53,7 @@ be compiled or run.
 * `ignore-pretty` will not compile the pretty-printed test (this is done to test the pretty-printer, but might not always work)
 * `ignore-test` always ignores the test
 * `ignore-lldb` and `ignore-gdb` will skip a debuginfo test on that debugger.
+* `ignore-gdb-version` can be used to ignore the test when certain gdb versions are used
```
Comment (nikomatsakis) [inline src/test/COMPILER_TESTS.md]: hmm I feel like this document ought to be removed and just become a pointer to

https://rust-lang-nursery.github.io/rustc-guide/tests/intro.html

(cc @mark-i-m, thoughts?)

Comment (nikomatsakis) [reply src/test/COMPILER_TESTS.md]: and this PR would be resubmitted at rustc-guide, natch

**Review by nikomatsakis — CHANGES_REQUESTED** [review]

File: src/test/COMPILER_TESTS.md (hunk 1 above)
Comment (nikomatsakis) [reply src/test/COMPILER_TESTS.md]: is there anything in here that is not present in the guide? I guess we have to check

Comment (tromey) [reply src/test/COMPILER_TESTS.md]: I looked and the guide seems like a superset of what's in this file.

**Comment by tromey** [comment]
https://github.com/rust-lang-nursery/rustc-guide/pull/179

**Comment by nikomatsakis** [comment]
Merged. @tromey would you object to repurposing this PR to make this a "soft link" over to the guide?

**Comment by tromey** [comment]
Not at all, I'll do it soon.

**Comment by tromey** [comment]
I don't think that failure is related to this patch.

**Comment by mark-i-m** [comment]
Thanks :)

**Pull request merged**